Sources/Sentry/Profiling/SentryProfiledTracerConcurrency.mm

```diff
@@ -16,19 +16,42 @@
 #    endif // SENTRY_HAS_UIKIT
 
 /**
- * a mapping of profilers to the tracers that started them that are still in-flight and will need to
- * query them for their profiling data when they finish. this helps resolve the incongruity between
- * the different timeout durations between tracers (500s) and profilers (30s), where a transaction
- * may start a profiler that then times out, and then a new transaction starts a new profiler, and
- * we must keep the aborted one around until its associated transaction finishes.
+ * a mapping of profilers to the number of tracers that started them that are still in-flight and
+ * will need to query them for their profiling data when they finish. this helps resolve the
+ * incongruity between the different timeout durations between tracers (500s) and profilers (30s),
+ * where a transaction may start a profiler that then times out, and then a new transaction starts a
+ * new profiler, and we must keep the aborted one around until its associated transaction finishes.
  */
 static NSMutableDictionary</* SentryProfiler.profileId */ NSString *,
-    NSMutableSet<SentryTracer *> *> *_gProfilersToTracers;
+    /* number of in-flight tracers */ NSNumber *> *_gProfilersToTracers;
 
 /** provided for fast access to a profiler given a tracer */
 static NSMutableDictionary</* SentryTracer.tracerId */ NSString *, SentryProfiler *>
     *_gTracersToProfilers;
 
+namespace {
+
+/**
+ * Remove a profiler from tracking given the id of the tracer it's associated with.
+ * @warning Must be called from a synchronized context.
+ */
+void
+_unsafe_cleanUpProfiler(SentryProfiler *profiler, NSString *tracerKey)
```

Comment:
`m`: The prefix `unsafe` raises a few questions for me at the first glance. I thought this function is somewhat dangerous 😄 . Actually it's just not thread safe. What about renaming it to `cleanUpProfiler_non_thread_safe`?
```suggestion
_cleanUpProfiler_non_thread_safe(SentryProfiler *profiler, NSString *tracerKey)
```